sql/core/src/main/scala/org/apache/spark/sql/execution/streaming/console.scala

```diff
@@ -49,7 +49,7 @@ class ConsoleSink(options: Map[String, String]) extends Sink with Logging {
     println("-------------------------------------------")
     // scalastyle:off println
     data.sparkSession.createDataFrame(
-      data.sparkSession.sparkContext.parallelize(data.collect()), data.schema)
```

Comment:
This might not be safe unfortunately.  Anything that changes the query plan might cause it to get replanned (and we don't do this correctly).

Can you make sure that there is a test case that uses the console sink and does aggregation across more than one batch?